src/test/java/generator/DecisionGeneratorTest.java

```diff
@@ -0,0 +1,72 @@
+package generator;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.stubbing.Answer;
+import project.*;
+
+import java.util.*;
+
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.Mockito.*;
+
+public class DecisionGeneratorTest {
+    private Map<Model.Side, TreeMap<Integer, Lane>> getExampleLaneMap(int leftLanes, int rightLanes){
+        Map<Model.Side, TreeMap<Integer, Lane>> lanesMap = new HashMap<>();
+        TreeMap<Integer, Lane> leftTreeMap = new TreeMap<>();
+        for(int i=1; i<=leftLanes; i++) {
+            leftTreeMap.put(i, null);
+        }
+        TreeMap<Integer, Lane> rightTreeMap = new TreeMap<>();
+        for(int i=1; i<=rightLanes; i++) {
+            rightTreeMap.put(i, null);
+        }
+        TreeMap<Integer, Lane> centerTreeMap = new TreeMap<>();
+        centerTreeMap.put(0, null);
+        lanesMap.put(Model.Side.LEFT, leftTreeMap);
+        lanesMap.put(Model.Side.RIGHT, rightTreeMap);
+        lanesMap.put(Model.Side.CENTER, centerTreeMap);
+        return lanesMap;
+    }
+
+    @Test
+    public void generateActionByDecisionMapTest() {
+        Map<Decision, Action> actionByDecision = new HashMap<>();
+        String baseIRI = "anything";
+        MyFactory factoryMock = mock(MyFactory.class);
+        Change_lane changeLaneMock = mock(Change_lane.class);
+        Model modelMock = mock(Model.class);
+        Scenario scenarioMock = mock(Scenario.class);
+
+        doNothing().when(changeLaneMock).addLane_change_by(anyInt());
```

Comment:
co robi doNothing()? Nie znam się na frameworku, to pytam

Reply:
Nic nie robi, po prostu, żeby przy addLange_change_by się program nie wywalał.